internal/experiments/experiments_ai_data_poisoning.go

```diff
@@ -0,0 +1,196 @@
+			Path:   "/ai-experiments",
+		}
+
+		var requestBody []byte
+		if &api.Payload != nil {
+			requestBody, err = json.Marshal(api.Payload)
+			if err != nil {
+				return err
+			}
+		}
+		fmt.Println(string(requestBody))
+		req, err := http.NewRequest("POST", url.String(), bytes.NewBuffer(requestBody))
+		if err != nil {
+			return err
+		}
+
+		req.Header.Add("Content-type", "application/json")
+
+		response, err := http.DefaultClient.Do(req)
+		if err != nil || response.StatusCode != 200 {
+			return err
+		}
+		defer response.Body.Close()
+		var apiResponse AIAPIResponse
+		err = json.NewDecoder(response.Body).Decode(&apiResponse)
+		if err != nil {
+			return err
+		}
+
+		results[api.Description] = ExecuteAIAPIResult{
+			Description:    api.Description,
+			ExperimentName: config.Metadata.Name,
+			Timestamp:      time.Now(),
+			Status:         response.StatusCode,
+			Response:       apiResponse,
+		}
+	}
+
+	resultJSON, err := json.Marshal(results)
+	if err != nil {
+		return fmt.Errorf("Failed to marshal experiment results: %w", err)
+	}
+
+	file, err := createTempFile(p.Type(), config.Metadata.Name)
+	if err != nil {
+		return fmt.Errorf("Unable to create file cache for experiment results %w", err)
+	}
+
+	_, err = file.Write(resultJSON)
+	if err != nil {
+		return fmt.Errorf("Failed to write experiment results: %w", err)
+	}
+
+	return nil
+}
+
+func (p *LLMDataPoisoningExperiment) Verify(ctx context.Context, client *k8s.Client, experimentConfig *ExperimentConfig) (*verifier.Outcome, error) {
+	var config LLMDataPoisoningExperiment
+	yamlObj, _ := yaml.Marshal(experimentConfig)
+	err := yaml.Unmarshal(yamlObj, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	v := verifier.New(
+		config.Metadata.Name,
+		config.Description(),
+		config.Framework(),
+		config.Tactic(),
+		config.Technique(),
+	)
+
+	rawResults, err := getTempFileContentsForExperiment(p.Type(), config.Metadata.Name)
+	if err != nil {
+		return nil, fmt.Errorf("Could not fetch experiment results: %w", err)
+	}
+
+	for _, rawResult := range rawResults {
+		var results map[string]ExecuteAIAPIResult
+		err = json.Unmarshal(rawResult, &results)
+		if err != nil {
+			return nil, fmt.Errorf("Could not parse experiment result: %w", err)
+		}
+
+		for _, api := range config.Parameters.Apis {
+			result, found := results[api.Description]
+			if !found {
+				continue
+			}
+			fail := false
+			if api.ExpectedResponse.VerifiedPromptChecks != nil {
+				for _, responseCheck := range api.ExpectedResponse.VerifiedPromptChecks {
+					if result.Response.VerifiedPromptChecks != nil {
+						for _, resultCheck := range result.Response.VerifiedPromptChecks {
+							if resultCheck.Check == responseCheck.Check {
+								if resultCheck.Detected != responseCheck.Detected {
+									fail = true
+									v.Fail(api.Description)
+								}
+							}
+						}
+					}
+				}
+			}
+			if !fail {
+				v.Success(api.Description)
```

Comment:
I had been doing the experiment name for this, as you could have multiple experiments output from the same file

Reply:
Not sure I understand. This is per AI API which made sense to me.

Reply:
I think I misread this, I see it's unique per instance now!